http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/DnsResolverLoader.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.nio.netty.internal;
+
+import io.netty.channel.socket.DatagramChannel;
+import io.netty.resolver.AddressResolverGroup;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.net.InetSocketAddress;
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+import software.amazon.awssdk.utils.internal.ClassLoaderHelper;
+
+/**
+ * Utility class for instantiating netty dns resolvers only if they're available on the class path.
+ */
+@SdkProtectedApi
+public class DnsResolverLoader {
+
+    private DnsResolverLoader() {
+    }
+
+    public static AddressResolverGroup<InetSocketAddress> init(Class<? extends DatagramChannel> channelType) {
+        try {
+            Class<?> addressResolver = ClassLoaderHelper.loadClass(getAddressResolverGroup(), false, (Class) null);
+            Class<?> dnsNameResolverBuilder = ClassLoaderHelper.loadClass(getDnsNameResolverBuilder(), false, (Class) null);
+
+            Object dnsResolverObj = dnsNameResolverBuilder.newInstance();
+            Method method = dnsResolverObj.getClass().getMethod("channelType", Class.class);
+            method.invoke(dnsResolverObj, channelType);
+
+            Object e = addressResolver.getConstructor(dnsNameResolverBuilder).newInstance(dnsResolverObj);
+            return (AddressResolverGroup<InetSocketAddress>) e;
```

Comment:
We don't specify a DNS Caching policy, which means [the default](https://netty.io/4.1/api/io/netty/resolver/dns/DefaultDnsCache.html) will be used.